src/identity/client.rs

```diff
@@ -141,8 +147,18 @@ impl Debug for PublicId {
 }
 
 impl Display for PublicId {
-    #[allow(trivial_casts)]
     fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
-        (self as &Debug).fmt(formatter)
+        let pub_id = unwrap!(bincode::serialize(&self));
+        write!(formatter, "{}", base64::encode(&pub_id))
+    }
+}
+
+impl FromStr for PublicId {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let pub_id = base64::decode(s).map_err(|e| Error::NetworkOther(e.to_string()))?;
```

Comment:
I wanted to make this conversion automatic by implementing the necessary traits but wasn't successful yet